src/parser/parser_support.c

```diff
@@ -2982,6 +2982,42 @@ pt_has_inst_or_orderby_num (PARSER_CONTEXT * parser, PT_NODE * node)
   return has_inst_orderby_num;
 }
 
+/*
+ * pt_has_inst_or_orderby_num_in_where ()
+ *          - check if tree has an INST_NUM or ORDERBY_NUM node in where
+ *   return: true if tree has INST_NUM/ORDERBY_NUM
+ *   parser(in):
+ *   node(in):
+ */
+
+bool
+pt_has_inst_or_orderby_num_in_where (PARSER_CONTEXT * parser, PT_NODE * node)
+{
+  bool has_inst_orderby_num = false;
+  PT_NODE *where;
+
+  switch (node->node_type)
+    {
+    case PT_SELECT:
+      where = node->info.query.q.select.where;
+      (void) parser_walk_tree (parser, where, pt_is_inst_or_orderby_num_node, &has_inst_orderby_num,
+			       pt_is_inst_or_orderby_num_node_post, &has_inst_orderby_num);
+      break;
+
+    case PT_UNION:
+    case PT_DIFFERENCE:
+    case PT_INTERSECTION:
+      has_inst_orderby_num |= pt_has_inst_or_orderby_num_in_where (parser, node->info.query.q.union_.arg1);
+      has_inst_orderby_num |= pt_has_inst_or_orderby_num_in_where (parser, node->info.query.q.union_.arg2);
+      break;
```

Comment:
the below query looks not handled properly
```
create table t1 (
	col1 int, col2 varchar(3), col3 int, col4 int
);

insert into t1 values (1, 'ABC', 2, 3), (2, 'BCD', 3, 4), (3, 'CDE', 4, 5), (4, 'DEF', 5, 6), (5, 'EFG', 6, 7);

create table t2 (
	col1 int, col2 varchar(3), col3 int, col4 int
);

insert into t2 values (5, 'ABC', 2, 3), (4, 'BCD', 3, 4), (3, 'CDE', 4, 5), (2, 'DEF', 5, 6), (1, 'EFG', 6, 7);


select /*+ recompile */ count(*)
from t1 t, ((select col1, col2, col3, col4 
		     from t2 where col4 > 0
			 union
	         select col1, col2, col3, col4 
		     from t1 where col4 > 1
		)
		limit 4
    ) s 
where t.col4 = s.col4 
  and s.col2 = 'ABC';
```

Reply:
I fixed it.